config.go

```diff
@@ -131,10 +131,9 @@ func (c *config) discoverExternalComponents(path string) error {
 	if err != nil {
 		return err
 	}
-	for plugin := range pluginPaths {
-		plugin := plugin
+	for plugin, path := range pluginPaths {
 		c.Builders[plugin] = func() (packer.Builder, error) {
-			return c.pluginClient(pluginPaths[plugin]).Builder()
+			return c.pluginClient(path).Builder()
```

Comment:
you probably want to do a `path := path` right at the beginning of the loop
otherwise path will refer to the last path.
see example:
https://play.golang.org/p/Ha3EdfQyr7D

Reply:
That is *really* weird, so basically you are saying if I had more than one external plugin, it would break?

I wondered why that assignment to itself was done, what is the deal with that?

Reply:
Right, thanks @yuval-k. I was wondering why they were doing that, must have something to do with the anonymous function? I *think* I understand it now :)

Reply:
yes, essentially the function captures the loop variable that gets changed in each iteration.
by doing `path := path`, the function will capture a copy of that variable.

Reply:
thanks :)